monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -580,9 +580,11 @@ sealed abstract class Task[+A] extends Serializable with TaskDeprecated.BinCompa
   @UnsafeBecauseImpure
   def runToFutureOpt(implicit s: Scheduler, opts: Options): CancelableFuture[A] = {
     val opts2 = opts.withSchedulerFeatures
-    Local.bindCurrentIf(opts2.localContextPropagation) {
-      TaskRunLoop.startFuture(this, s, opts2)
-    }
+    Local
+      .bindCurrentAsyncIf(opts2.localContextPropagation) {
+        TaskRunLoop.startFuture(this, s, opts2)
+      }
+      .asInstanceOf[CancelableFuture[A]]
```

Comment:
I hope there is a way to do it without it

Reply:
I don't think there is. But `bind..If` is somewhat of internal method, we can make a version for `CancelableFuture` and make it private.

Reply:
Hmm, maybe it could make sense to make `bindCurrentIf` private as well? It's basically `if p isolate(x) else x` so if anyone needs it - he can add it easily

Reply:
Yes, that what I was thinking of :)